reformCage.py

```diff
@@ -0,0 +1,177 @@
+parser.add_argument('mol_file')
+parser.add_argument('-c', '--reform_cage',
+                    action='store_true',
+                    help=('Boolean: writes the unrelaxed cage to a .mol file.'))
+parser.add_argument('-b', '--write_building_blocks',
+                    action='store_true',
+                    help=('Boolean: writes the unrelaxed building-blocks to a .mol file.'))
+args = parser.parse_args()
+name = str(args.mol_file)[0:-4]
+
+def reform_cage(self, deconstructed_topology):
+  """
+  reform the unrelaxed stk cage.
+
+  """
+  building_blocks = []
+  for k in self:
+    bb_mol = AllChem.MolFromSmiles(k)
+    bb_Hs = AllChem.AddHs(bb_mol)
+    AllChem.EmbedMolecule(bb_Hs)
+    if self[k][0] >= 3:
+      bb_su = stk.StructUnit3(bb_Hs, [self[k][1]])
+      if bb_su not in building_blocks:
+        building_blocks.append(bb_su)
+      else:
+        pass
+    else:
+      bb_fg = [self[k][1]]
+      bb_su = stk.StructUnit2(bb_Hs, [self[k][1]])
+      if bb_su not in building_blocks:
+        building_blocks.append(bb_su)
+      else:
+        pass
+
+  cage = stk.Cage(building_blocks, deconstructed_topology)
+  number_of_bonds_made = cage.bonds_made
+
+  cage.write(name+'-out.mol')
+  cage.dump(name+'-out.json')
+  return cage
+
+def write_building_blocks(self):
+  """
+  writes the optimised building blocks to .mol file
+
+  """
+  wbb_count = 1
+  for k in self:
+    bb_mol = AllChem.MolFromSmiles(k)
+    bb_Hs = AllChem.AddHs(bb_mol)
+    AllChem.EmbedMolecule(bb_Hs)
+    if self[k][0] >= 3:
+      bb_su = stk.StructUnit3(bb_Hs, [self[k][1]])
+      bb_su.write(name+'-'+str(wbb_count)+'bb.mol')
+    else:
+      bb_su = stk.StructUnit2(bb_Hs, [self[k][1]])
+      bb_su.write(name+'-'+str(wbb_count)+'bb.mol')
+    wbb_count += 1
+
+def topology_calc(self):
+  """
+  finds the topology of the system.
+
+  """
+  if str(self) == '[(2, 3), (3, 2)]':
+    return stk.TwoPlusThree()
+  elif str(self) == '[(2, 6), (3, 4)]':
+    return stk.FourPlusSix()
+  elif str(self) == '[(2, 9), (3, 6)]':
+    return stk.SixPlusNine()
+  elif str(self) == '[(2, 12), (3, 8)]':
+    return stk.EightPlusTwelve()
+  elif str(self) == '[(3, 4), (4, 2)]':
+    return stk.TwoPlusFour()
+  elif str(self) == '[(3, 6), (4, 3)]':
+    return stk.ThreePlusSix()
+  elif str(self) == '[(3, 12), (4, 6)]':
+    return stk.SixPlusTwelve()
+  else:
+    return 'unknown'
+
+def main():
+  mol = AllChem.MolFromMolFile(args.mol_file)
+  imine = AllChem.MolFromSmiles('CC=NC')
+  subs = mol.GetSubstructMatches(imine)
+
+  bonds = []
+  for i in subs:
+    bonds.append(i[1:3])
+  bond_ids = [mol.GetBondBetweenAtoms(x,y).GetIdx() for x,y in bonds]
+  new_mol = AllChem.FragmentOnBonds(mol,bond_ids)
+
+  new_smiles = AllChem.MolToSmiles(new_mol, True)
+  new_split = new_smiles.split('.')
+
+  building_blocks_dict = {}
+  smiles_fragments = []
+  uncorrected_smiles_fragments = []
+  unknown_functional_group = AllChem.MolFromSmiles('*')
+
```

Comment:
The various if, else block in this function are quite big, I recommend making them into separate functions.